docs/guides/1230-persistent-cache.md

```diff
@@ -0,0 +1,220 @@
+package ci
+
+import (
+  "dagger.io/dagger"
+  "universe.dagger.io/go"
+)
+
+dagger.#Plan & {
+  // Retrieve source from current directory
+  // Input
+  client: filesystem: ".": read: {
+    contents: dagger.#FS
+    // Include only contents we need to build our project
+    // Any other files or patterns can be added
+    include: ["**/*.go", "go.mod", "go.sum"]
+  }
+
+  // Output
+  client: filesystem: "./output": write: {
+    contents: actions.build.output
+  }
+
+  actions: {
+    // Alias on source
+    _source: client.filesystem.".".read.contents
+
+    // Build go binary
+    build: go.#Build & {
+      source: _source
+    }
+  }
+}
+```
+
+To build a Go binary, just run `dagger do build`. It should then take some time to
+install dependencies, build the binary and output it.
+
+Here's an example of a run
+
+```shell
+[✔] actions.build.container                                               338.0s
+[✔] client.filesystem.".".read                                              0.1s
+[✔] actions.build.container.export                                          0.1s
+[✔] client.filesystem."./output".write                                      0.2s
+```
+
+Indeed, Dagger has an ephemeral cache so if you rerun it, that shouldn't take
+that long.
+
+```shell
+[✔] actions.build.container                                                 2.9s
+[✔] client.filesystem.".".read                                              0.0s
+[✔] actions.build.container.export                                          0.0s
+[✔] client.filesystem."./output".write                                      0.1s
+```
+
+But if you stop the Buildkit daemon and remove its storage volume, cache will be lost and
+all actions will be executed again.
+
+:::info
+Now we have seen how ephemeral cache works, let's continue to understand how
+store cache in your local filesystem, so you can clean your Buildkit daemon without
+losing all your CI's cache.
+:::
+
+## Persistent cache in a remote registry
+
+Buildkit can also import/export cache to a registry. This is a great way to
+share cache between your team and avoid flooding your filesystem.
+
+:::tip
+Using a registry as cache storage is more efficient than local storage because Buildkit will only re-export
+missing layers on multiple runs.
+:::
+
+To store cache in an external registry, you just need to add flags to dagger command :
+`dagger do <action> --cache-to type=registry,mode=max,ref=<registry target>/<image> --cache-from type=registry,ref=<registry target>/<image>`
+
+:::tip
+Using `mode=max` argument will cache **all** layers from intermediate
+steps, which is really useful in the context of Dagger where you will have
+multiple steps to execute.  
+To only store the final layers of the exported result, use `mode=min`.
+:::
+
+Let's first deploy a simple registry in your localhost
+
+```shell
+docker run -d -p 6000:5000 --restart=always --name cache-registry registry:2
```

Comment:
```suggestion
docker run -d -p 5000:5000 --restart=always --name cache-registry registry:2
```

There is a port mismatch between this line and line 121.
Either this value needs to be changed to 5000 or line 121 needs to change to 6000

Reply:
Resolved in https://github.com/dagger/dagger/pull/2035/commits/ebac13380004ecaa71f6a37dcd68e02ffad71943. Thanks @tdjordan!